### PR TITLE
03 - Container Registry - Shipping out!

### LINKED ISSUE #4: 03 - container registry - Shipping out!
Building an image and Running a container works great!
    But.. it scales poorly. Imagine having a plethora of servers?
    You'd have to build it everywhere,
    which is a fun full-time job, if you think repetitive work is fun!
    But alas, that's not what you were hired to do.

Luckily you've heard about DockerHub,
    and your application is open source or "public" anyways,
    so there's no harm in storing it in a public repo.

### Task

- Augment your Workflow so it's pushing to DockerHub.
    - It probably needs some credentials for this.
- Your deploy step probably just runs the image,
    which means it uses the one that's available locally.
    Change this so it explicitly pulls your newly built image first.

### PR DESCRIPTION
**Pull request by devantler**
Added a workflow that tests the frontend and backend in parallel.

### CODE REVIEW

File: .github/workflows/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: deployment
+on:
+  push:
+    branches:
+      - "trunk"
+env:
+  COMMIT_SHA: ${{ github.event.head.sha }}
+jobs:
+  build-and-push-frontend-image:
+    runs-on: ubuntu-latest
```
Comment (Hounsvad) [inline .github/workflows/deployment.yaml:10]: Use specific version for stability

File: .github/workflows/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/{ github.repository_owner }/cookie-frontend
```
Comment (Hounsvad) [inline .github/workflows/deployment.yaml]: Use repository name as well

File: .github/workflows/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        uses: docker/build-push-action@v3
+        with:
+          context: frontend
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+  build-and-push-backend-image:
+    runs-on: ubuntu-latest
```
Comment (Hounsvad) [inline .github/workflows/deployment.yaml:48]: Use specific version for stability

File: .github/workflows/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/{ github.repository_owner }/cookie-backend
```
Comment (Hounsvad) [inline .github/workflows/deployment.yaml]: Use repository name as well

**Review by Hounsvad — CHANGES_REQUESTED** [review]
Add repo association to dockerfiles

File: .github/workflows/deployment.yaml (hunk 1 above)
Comment (devantler) [reply .github/workflows/deployment.yaml:10]: I agree in general, but for this project idc

File: .github/workflows/deployment.yaml (hunk 4 above)
Comment (devantler) [reply .github/workflows/deployment.yaml]: resolved

File: .github/workflows/deployment.yaml (hunk 3 above)
Comment (devantler) [reply .github/workflows/deployment.yaml:48]: idc

File: .github/workflows/deployment.yaml (hunk 2 above)
Comment (devantler) [reply .github/workflows/deployment.yaml]: resolved

**Review by harpunfiskeren — APPROVED** [review]
Looks good

**Review by Hounsvad — APPROVED** [review]

**Pull request merged**